From 222bf3038db688fa619bc335a71bc6694844b62d Mon Sep 17 00:00:00 2001
From: Aniruddha <aniruddhac@nvidia.com>
Date: Mon, 22 Oct 2012 16:24:34 +0530
Subject: [PATCH 2/6] [SF Streaming] Trigger reconnect if needed

In NuCachedSource2 mNumRetriesLeft is greater than 0
then dont return mFinalStatus until reconnection attemps
fail.

Bug 1050863

Change-Id: I08b5dd52e6e69fb8abd7e0fa7736b4f6338fedce
---
 media/libstagefright/NuCachedSource2.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/NuCachedSource2.cpp b/media/libstagefright/NuCachedSource2.cpp
index 4eff59a..855fa07 100644
--- a/media/libstagefright/NuCachedSource2.cpp
+++ b/media/libstagefright/NuCachedSource2.cpp
@@ -632,7 +632,7 @@ ssize_t NuCachedSource2::readInternal(off64_t offset, void *data, size_t size) {
 
     size_t delta = offset - mCacheOffset;
 
-    if (mFinalStatus != OK) {
+    if (mFinalStatus != OK && mNumRetriesLeft == 0) {
         if (delta >= mCache->totalSize()) {
             return mFinalStatus;
         }
-- 
1.7.1

